src/digest.rs

```diff
@@ -448,8 +447,66 @@ pub extern fn SHA512_4(out: *mut u8, out_len: c::size_t,
     polyfill::slice::fill_from_slice(out, digest);
 }
 
+unsafe extern fn sha1_block_data_order(state: *mut u64, data: *const u8, num: c::size_t) {
```

Comment:
See https://github.com/briansmith/ring/commit/279ca8e1b0e0f4eef93c0c5ab458671b095ec41f, which demonstrates how to add a (private) ring::digest::sha1 submodule.